### ISSUE TITLE
org-mode support

### ISSUE THREAD

**Issue by orissus** [zed-industries/zed#11138]
### Check for existing issues

- [X] Completed

### Describe the feature

This is a hopeful ask to put into consideration adding *some* support for [Org mode](https://en.wikipedia.org/wiki/Org-mode). 

For example: 
- implementing org file format awareness (headings, links, tags etc) 
- implementing collapsible headings (under cursor and for all) 
- Making built-in`journal` format a choice between Markdown or Org-mode.

Loving Zed, with love, 
O.

**Comment by JohnCarterGonzalez** [comment]
this would be amazing. Long term would be cool to integrate similar functionality to org-roam

**Comment by opsnull** [comment]
this would be amazing, the only reason I persist to emacs.

**Comment by reynard93** [comment]
any updates for this? i see we are getting pretty close with the repl support

**Comment by notpeter** [comment]
@hron Just added an `org` zed extension which adds syntax highlighting support for Org mode files.  It's a ways from a complete org-mode implementation, but it's a start:
- https://github.com/zed-industries/extensions/pull/1617

**Comment by notpeter** [comment]
No, @Sepheriel, this is not currently possible.  It could be possible to implement some additional functionality (link navigation, auto complete) via a dedicated org-mode LSP and connect that to Zed.  But it would not support rendering generated pages (e.g. templates, custom font style for headings, bullets, etc) nor rendering pages with generated content (e.g. backlinks, TOC, etc).

**Comment by Sepheriel** [comment]
@notpeter thanks for the reply! Too bad, then i'll wait and see if any changes to the API will make it possible in the future before putting any work in the idea :/
